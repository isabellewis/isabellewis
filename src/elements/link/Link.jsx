import React from 'react';
import {
  LinkA
} from './Link.style';

export const Link = ({
  link,
  linkClass,
  label
}) => (
	<LinkA href={link} className={linkClass} target='_blank'>
		{ label }
	</LinkA>
);