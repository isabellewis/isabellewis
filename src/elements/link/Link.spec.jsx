import React from 'react';
import { shallow } from 'enzyme';
import { Link } from './Link';

describe('elements/Link', () => {
	let component;

	beforeEach(() => {
		component = shallow(<Link label='Link test text' />);
	});

	it('should match the snapshot', () => {
		expect(component).toMatchSnapshot();
	});
});