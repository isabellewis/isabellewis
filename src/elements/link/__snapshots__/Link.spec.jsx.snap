// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elements/Link should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Link
    label="Link test text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Link test text",
      "className": undefined,
      "href": undefined,
      "target": "_blank",
    },
    "ref": null,
    "rendered": "Link test text",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Link test text",
        "className": undefined,
        "href": undefined,
        "target": "_blank",
      },
      "ref": null,
      "rendered": "Link test text",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
